strictdoc/backend/sdoc/models/type_system.py

```diff
@@ -31,55 +31,127 @@ class RequirementFieldType:
     TYPE_VALUE = "TypeValue"
 
 
+class GrammarReferenceType:
+    PARENT_REQ_REFERENCE = "ParentReqReference"
+    FILE_REFERENCE = "FileReference"
+
+
 class ReferenceType:
     PARENT = "Parent"
     FILE = "File"
 
+    GRAMMAR_REFERENCE_TYPE_MAP = {
+        PARENT: GrammarReferenceType.PARENT_REQ_REFERENCE,
+        FILE: GrammarReferenceType.FILE_REFERENCE,
+    }
+
 
 class GrammarElementField:
     def __init__(self):
         self.title: str = ""
+        self.gef_type: str = ""
         self.required: bool = False
 
 
 class GrammarElementFieldString(GrammarElementField):
-    def __init__(self, parent, title: str, required: str):
+    def __init__(self, parent, title: str, required: Optional[str]):
```

Comment:
Is there a reason why `required` is set to Optional here? In the grammar, the `REQUIRED` is always provided for a grammar field, so here I would prefer it strict as well.

Reply:
Yes, you are right.
Removed the Optional
 `required: Optional[str]` => `required: str`